marklogic-data-hub/src/test/java/com/marklogic/hub/impl/GenerateIndexesTest.java

```diff
@@ -0,0 +1,142 @@
+package com.marklogic.hub.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.ApplicationConfig;
+import com.marklogic.hub.HubTestBase;
+import com.marklogic.hub.entity.DefinitionType;
+import com.marklogic.hub.entity.DefinitionsType;
+import com.marklogic.hub.entity.HubEntity;
+import com.marklogic.hub.entity.PropertyType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = ApplicationConfig.class)
+public class GenerateIndexesTest extends HubTestBase {
+
+    private static final String CODEPOINT = "http://marklogic.com/collation/codepoint";
+    private static final String ROOT = "http://marklogic.com/collation/";
+
+    private List<JsonNode> entities = new ArrayList();
+    private ObjectNode indexes;
+
+    @Test
+    public void sharedPropertyWithNullNamespace() {
+        final String namespace = null;
+        givenAnEntityWithTitleProperty("Book", namespace, CODEPOINT);
```

Comment:
I often like using given/when/then methods in a JUnit test to mirror what ideally will be in the JIRA ticket.